internal/core/src/common/RegexQuery.cpp

```diff
@@ -15,21 +15,50 @@
 #include "common/RegexQuery.h"
 
 namespace milvus {
+
+bool
+is_special(char c) {
+    // initial special_bytes_bitmap only once.
+    static std::string special_bytes(R"(\.+*?()|[]{}^$)");
+    static char special_bytes_bitmap[16];
+    static bool _ = []() -> bool {
+        for (char b : special_bytes) {
+            special_bytes_bitmap[b % 16] |= 1 << (b / 16);
+        }
+        return true;
+    };
+    return special_bytes_bitmap[c % 16] & (1 << (c / 16)) != 0;
+}
+
+std::string
+quote_meta(const std::string& s) {
+    size_t i;
+    size_t l = s.length();
+    std::string r;
+    for (i = 0; i < l; i++) {
+        if (is_special(s[i])) {
+            r += '\\';
+        }
+        r += s[i];
+    }
+    return r;
+}
+
 std::string
-ReplaceUnescapedChars(const std::string& input,
-                      char src,
-                      const std::string& replacement) {
+replace_unescaped_chars(const std::string& input,
+                        char src,
+                        const std::string& replacement) {
     std::string result;
-    bool escapeMode = false;
+    bool escape_mode = false;
 
     for (char c : input) {
-        if (escapeMode) {
+        if (escape_mode) {
             result += '\\';
             result += c;
-            escapeMode = false;
+            escape_mode = false;
         } else {
             if (c == '\\') {
-                escapeMode = true;
+                escape_mode = true;
             } else if (c == src) {
                 result += replacement;
```

Comment:
too much string concatenation.

Reply:
Should replace `%` but not `\\%` with `[\\s\\S]*`?
But current implementation did't do this at my view.

Reply:
> too much string concatenation.

@longjiquan   use strings.Builder to avoid frequent concatenation.
refer to 
```
        var b strings.Builder
        b.Grow(newStringLength)
        b.WriteString(`"`)
        b.String()
```

Reply:
> too much string concatenation.
use std::stringstream ?